string/string.mbt

```diff
@@ -118,36 +118,234 @@ pub fn iter(self : String) -> Iter[Char] {
   )
 }
 
+/// Removes the around whitespaces of this string.
+pub fn trim(self : String) -> String {
+  let len = self.length()
+  let mut start = 0
+  let mut end = len
+  while start < end && self[start] <= ' ' {
+    start += 1
+  }
+  while start < end && self[end - 1] <= ' ' {
+    end -= 1
+  }
+  if start > 0 || end < len {
+    self.substring(~start, ~end)
+  } else {
+    self
+  }
+}
+
+/// Returns true if this string is empty.
+pub fn is_empty(self : String) -> Bool {
+  self == ""
+}
+
+/// Returns true if this string is blank.
+pub fn is_blank(self : String) -> Bool {
+  self.trim() == ""
+}
+
+/// Returns the first index of the sub string.
+pub fn index_of(self : String, str : String, ~from : Int = 0) -> Int {
+  let len = self.length()
+  let sub_len = str.length()
+  let max_idx = len - sub_len
+  if sub_len == 0 {
+    0
+  } else if sub_len > len {
+    -1
+  } else {
+    let first = str[0]
+    let mut i = from
+    while i <= max_idx {
+      // find first character
+      while i < len && self[i] != first {
+        i += 1
+      }
+      // check the rest
+      if i <= max_idx {
+        let mut j = i + 1
+        let mut k = 1
+        let end = j + sub_len - 1
+        while j < end && self[j] == str[k] {
+          j += 1
+          k += 1
+        }
+        if j == end {
+          return i
+        }
+      }
+      i += 1
+    }
+    -1
+  }
+}
+
+/// Returns the last index of the sub string.
+pub fn last_index_of(
+  self : String,
+  str : String,
+  ~from : Int = self.length()
```

Comment:
should `last_index_of` and `index_of` be fused to a single API, e.g,
```
index_of(self : String, str : String, ~rev : Bool = true, ~from = if rev { self.length()} else {0})
```

cc @Guest0x0